File: lib/models/fdroid_app.dart
class FDroidApp {
  final String packageName;
  final String name;
  final String summary;
  final String description;
  final String iconUrl;
  final String version;
  final int versionCode;
  final int size;
  final String apkUrl;
  final String license;
  final String category;
  final String author;
  final String website;
  final String sourceCode;
  final DateTime added;
  final DateTime lastUpdated;
  final List<String> screenshots;
  final List<String> antiFeatures;
  final int downloads;
  final bool isInstalled;
  final String repository;
  
  FDroidApp({
    required this.packageName,
    required this.name,
    required this.summary,
    required this.description,
    required this.iconUrl,
    required this.version,
    this.versionCode = 1,
    required this.size,
    required this.apkUrl,
    required this.license,
    required this.category,
    required this.author,
    required this.website,
    required this.sourceCode,
    required this.added,
    required this.lastUpdated,
    required this.screenshots,
    this.antiFeatures = const [],
    required this.downloads,
    required this.isInstalled,
    this.repository = 'F-Droid',
  });
}

File: app/src/main/java/app/flicky/data/model/FDroidApp.kt
package app.flicky.data.model

import androidx.room.Entity
import androidx.room.Index
import androidx.room.PrimaryKey
import kotlinx.serialization.Serializable

@Serializable
@Entity(
    tableName = "apps",
    indices = [
        Index(value = ["name"]),
        Index(value = ["summary"]),
        Index(value = ["packageName"]),
        Index(value = ["category"])
    ]
)
data class FDroidApp(
    @PrimaryKey val packageName: String,
    val name: String,
    val summary: String,
    val description: String,
    val iconUrl: String,
    val version: String,
    val versionCode: Int,
    val size: Long,
    val apkUrl: String,
    val license: String,
    val category: String,
    val author: String,
    val website: String,
    val sourceCode: String,
    val added: Long,
    val lastUpdated: Long,
    val screenshots: List<String> = emptyList(),
    val antiFeatures: List<String> = emptyList(),
    val downloads: Long = 0,
    val isInstalled: Boolean = false,
    val repository: String = "F-Droid",
    val sha256: String = "",
    val whatsNew: String = ""
)
